src/cropduster.js

```diff
@@ -1,189 +1,183 @@
-    a = a || document.createElement('a');
+  setClickthrough(url) {
+    const a = document.getElementById('mi_dynamic_link') || document.createElement('a');
+
     a.href = url;
-    a.id = "mi_dynamic_link";
-    a.style.display = "none";
+    a.id = 'mi_dynamic_link';
+    a.style.display = 'none';
+
     document.body.appendChild(a);
 
     return a;
   },
 
-  setExtraData: function(dataObject) {
-    var el = document.querySelector("#mi-data");
-    el = el || document.createElement('div');
-    el.id = "mi-data";
-    el.style.display = "none";
+  setExtraData(dataObject) {
+    const el = document.getElementById('mi-data') || document.createElement('div');
 
-    var existingData;
+    el.id = 'mi-data';
+    el.style.display = 'none';
+
+    let existingData;
     try {
       existingData = JSON.parse(el.getAttribute('data-mi-data')) || {};
-    } catch(e) {
+    } catch (_) {
       // Overwrite if there was something in mi-data that wasn't JSON
       existingData = {};
     }
 
-    for(var i in dataObject) {
-      if(dataObject.hasOwnProperty(i)) {
-        existingData[i] = dataObject[i];
+    for (const key in dataObject) {
+      if (dataObject.hasOwnProperty(key)) {
+        existingData[key] = dataObject[key];
       }
     }
+
     el.setAttribute('data-mi-data', JSON.stringify(existingData));
     document.body.appendChild(el);
 
     return el;
   },
 
-  proxyUrl: function(url) {
-    var a = document.createElement('a');
-    var port = "";
+  proxyUrl(url) {
+    const a = document.createElement('a');
     a.href = url;
 
-    if (a.port === '0' || a.port === "") {
-      port = a.protocol == "https:" ? ":443" : "";
+    let port = '';
+    if (a.port === '0' || a.port === '') {
+      port = a.protocol === 'https:' ? ':443' : '';
     } else {
-      port = ":" + a.port;
+      port = `:${a.port}`;
     }
 
-    return [
-      CD.CORS_PROXY_SERVER,
-      "/",
-      a.hostname,
-      port,
-      a.pathname,
-      a.search,
-      a.hash
-    ].join("");
+    const { hostname, pathname, search, hash } = a;
+    return `${CD.CORS_PROXY_SERVER}/${hostname}${port}${pathname}${search}${hash}`;
   },
 
   // internal, do not modify
   _readyToCapture: true,
 
-  _openCalls: 0,
-
-  _reset: function() {
-    CD._openCalls = 0;
+  _reset() {
     CD._readyToCapture = true;
   },
 
-  suspend: function(maxSuspension, msg) {
-    msg = msg || "manual suspension";
+  pause(maxSuspension, msg) {
+    msg = msg || 'manual suspension';
 
-    if(maxSuspension) {
-      msg += ", will end in " + maxSuspension + "ms";
+    if (maxSuspension) {
+      msg += `, will end in ${maxSuspension}ms`;
 
-      setTimeout(function() {
-        CD.capture(msg);
+      setTimeout(() => {
+        MICapture.resume(msg);
       }, maxSuspension);
     }
 
-    CD._openCalls++;
-    CD._readyToCapture = false;
-    if (typeof(MICapture) == 'undefined') {
-      CD.log("suspended: " + msg);
+    if (typeof MICapture === 'undefined') {
```

Comment:
This pattern is used a lot, would it make sense to extract it into its own method?